### ISSUE TITLE
[ENH] Fix failing output tests for the ClusterSegmenter detector

### ISSUE THREAD

**Issue by alyssadsouza** [sktime/sktime#7462]
## Summary
The following tests for the ClusterSegmenter detector fail and thus have been added to the list of excluded tests:
- `test_predict_points`
- `test_predict_segments`
- `test_transform_output_type`

We should fix ClusterSegmenter to make sure these tests pass.

## Related PR
The tests were added in #6958 and updated in #7459.

**Comment by Spinachboul** [comment]
@alyssadsouza 
Can I be assigned this issue
I had been working in the past under @Alex-JG3 to solve for the failing tests for `GreedyGaussianSegmentation`
Thanks!

**Comment by alyssadsouza** [comment]
@Spinachboul yup, go for it 👍 I think either @Alex-JG3 or @fkiraly will need to assign you as I don't have that permission

**Comment by Spinachboul** [comment]
@alyssadsouza 
Coud you please provide the scipt for the test, and also the logs of errors for the failing tets if possible?
That would give some idea of where the issue is occuring!

**Comment by Spinachboul** [comment]
@fkiraly & @Alex-JG3 

I encountered an error while running the tests that seems to originate from the `_generate_estimator_instance` function in `_conditional_fixtures.py`. The error is raised because the object returned by `_generate_estimator_instance` does not have a `clone` method, which appears to be a requirement for subsequent operations in the code.

---

### **Details of the Error**:
1. **Traceback**:
   - The traceback points to the `get_fixtures` function, where `generator_dict[fixture_var](test_name, **kwargs)` is called.
   - The object returned by this generator function does not have a `clone` method, leading to an `AttributeError`.

2. **Potential Cause**:
   - `_generate_estimator_instance` is returning an object that is not a valid estimator or does not meet the expected requirements (e.g., having a `clone` method).

3. **Steps I Tried**:
   - I inspected `_generate_estimator_instance` but could not identify why the generated object lacks the `clone` method.
   - I added debug statements to check the output of `generator_dict[fixture_var]`, and it seems to generate unexpected or incomplete objects in certain cases.
   - I considered filtering out objects without `clone`, but this might not address the root issue and could lead to incomplete test coverage.

4. **Relevant Code**:
   - `_generate_estimator_instance` is responsible for producing the estimator, and the error occurs when its result is used in `get_fixtures`.

---

### **What I Need Help With**:
1. **Clarification**:
   - What are the expected outputs of `_generate_estimator_instance`? Should all generated objects inherently support `clone`?
   - Are there specific cases where `_generate_estimator_instance` is known to fail, and should these cases be explicitly handled or skipped?
   - How can I ensure `_generate_estimator_instance` generates only valid objects? Should I modify its logic or the logic in `get_fixtures` to handle such cases?
   - Is skipping faulty estimators (e.g., objects without `clone`) an acceptable workaround, or does this indicate a deeper issue in the generation process?
   - Should additional checks or logging be added to `_generate_estimator_instance` to identify why invalid objects are being produced?

**Comment by Alex-JG3** [comment]
I'll do my best to answer your questions although I am not too familar with these parts of the codebase.

> What are the expected outputs of _generate_estimator_instance? Should all generated objects inherently support clone?

Get all objects should support clone since they should all inherit from `BaseEstimator` which must implement clone.

https://github.com/sktime/skbase/blob/8bfa0f4d8707dd4f148039f92de4ff32500e602c/skbase/base/_base.py#L157

> Are there specific cases where _generate_estimator_instance is known to fail, and should these cases be explicitly handled or skipped?

I am afraid I cannot answer this one. Someone with a wider knowledge of the codebase may be able to answer this.

> How can I ensure _generate_estimator_instance generates only valid objects? Should I modify its logic or the logic in get_fixtures to handle such cases?

I would first ensure that the estimator itself is implemented correctly. If there is an estimator that does not have a clone method, that is an issue. If you could share the estimator in question that would be very helpful.

> Is skipping faulty estimators (e.g., objects without clone) an acceptable workaround, or does this indicate a deeper issue in the generation process?

Similar to the previous point. We should make sure the estimator itself is correctly implemented since it should have a clone method.

> Should additional checks or logging be added to _generate_estimator_instance to identify why invalid objects are being produced?

What do you suggest? Would checking if estimators from `_generate_estimator_instance` inherit from correct base help?
